### ISSUE TITLE
Psalm expects first parameter of strval to not be mixed

### ISSUE THREAD

**Issue by davidsneighbour** [vimeo/psalm#4201]
https://psalm.dev/r/179a7d3437

```
class TransformerEmail
{
    /**
     * @param mixed $value
     *
     * @return string
     */
    public static function create($value = null)
    {
        $value = strval($value);
        return strtolower($value);
    }
}
```

https://www.php.net/manual/en/function.strval.php states that the first parameter of strval is mixed. Psalm says "Argument 1 of strval cannot be mixed|null, expecting null|object|scalar".

Do I miss something or is this a bug?

**Comment by weirdan** [comment]
`strval()` produces warnings when given an array (https://3v4l.org/GVsej), indicating that it's not supposed to safely accept all of the types that `mixed` comprises.

**Comment by davidsneighbour** [comment]
You are right. The following changes stops the error from coming up.

https://psalm.dev/r/ebfbed9d79

The only problem I would have is that in PHP7.4 (my target system) this is only a Notice. In PHP 8.0 they changed it to Warning and as I read they changed the documentation for many functions to show proper typecasting. I wonder where they hide that documentation.

I am closing this issue now. Using `null|object|scalar` works for me.

**Comment by davidsneighbour** [comment]
on a sidenote: `(string)$value` does not throw errors.
